test/rpc_test.py

```diff
@@ -919,6 +929,34 @@ def test_remote_same_worker(self):
         )
         self.assertEqual(rref_c.to_here().wait(), torch.ones(n, n) + 4)
 
+    @dist_init(setup_model_parallel=True)
+    def test_call_method_on_rref(self):
+        """
+        Tests that it is possible to call an instance method on a remote objet
+        by using rref.owner() as destination of the call.
+        """
+        vals = [10, 2, 5, 7]
+        dst_rank = (self.rank + 1) % self.world_size
+        dst_worker = "worker{}".format(dst_rank)
+
+        # creates a remote object
+        rref = rpc.remote(dst_worker, MyClass, args=(vals[0], ))
+
+        # modifies state of the remote object
+        rpc.rpc_sync(rref.owner(), _call_method_on_rref, args=(
```

Comment:
The docs for `rpc_sync` say: `to (int or str): id or name of the destination worker.`, but it seems like we allow `WorkerInfo` here? Can we avoid supporting `WorkerInfo` in the RPC API and keep it to string/int? This way it would be easier to torchscript this API (otherwise we need to make WorkerInfo a torchscript type).

Reply:
Yep, the doc is deprecated and should be updated, it is now either string or workerinfo

Reply:
I thought we will have a different API for TorchScript APIs in `distributed.rpc.jit.rpc_sync/rpc_async/remote`?

Reply:
How about we land this PR as is -- the unit tests are still going to be relevant regardless of us passing in int or WorkerInfo. Then, in a separate PR, refactor both RRef.owner() and the other APIs to take int instead of WorkerInfo.

Reply:
> I thought we will have a different API for TorchScript APIs in distributed.rpc.jit.rpc_sync/rpc_async/remote?

Yes, although at some point we would like to merge this into a single API once jit supports Callables. If we do that, then we need to support WorkerInfo as a torchscript type as well.

Reply:
> Then, in a separate PR, refactor both RRef.owner() and the other APIs to take int instead of WorkerInfo.

That sounds good to me. We are going to quite some API changes any way.